zerver/actions/realm_settings.py

```diff
@@ -552,3 +560,73 @@ def do_send_realm_reactivation_email(realm: Realm, *, acting_user: Optional[User
+
+
+def do_send_realm_deactivation_email(realm: Realm, acting_user: Optional[UserProfile]) -> None:
+    deactivation_date = timezone_now().date()
+    shared_context = {
+        "realm_name": realm.name,
+        "event_date": deactivation_date,
+        "corporate_enabled": settings.CORPORATE_ENABLED,
+    }
+    language = realm.default_language
+    owner_emails = set(realm.get_human_owner_users().values_list("delivery_email", flat=True))
+
+    not_deactivated_by_owner = False
+
+    # This is an optional parameter for the deactivate_realm management command;
+    # there is no deactivating user name for the email to owners.
+    if acting_user is None:
+        not_deactivated_by_owner = True
+
+    # This is realm deactivation is from the support panel;
+    # we do not share the deactivating user's name in this case.
+    if acting_user is not None and acting_user.delivery_email not in owner_emails:
+        not_deactivated_by_owner = True
+
+    if not_deactivated_by_owner:
+        not_owner_context = shared_context.copy()
+        not_owner_context["no_acting_user"] = True
+        send_email(
+            "zerver/emails/realm_deactivated",
+            to_emails=list(owner_emails),
+            from_name=FromAddress.security_email_from_name(language=language),
+            from_address=FromAddress.tokenized_no_reply_address(),
+            language=language,
+            context=not_owner_context,
+            realm=realm,
+        )
+    else:
+        assert acting_user is not None
+        owner_emails.discard(acting_user.delivery_email)
+        shared_context["no_acting_user"] = False
+
+        # We adjust the email text for the organization owner
+        # who did the realm deactivation.
+        acting_owner_context = shared_context.copy()
+        acting_owner_context["is_acting_owner"] = True
+        send_email(
+            "zerver/emails/realm_deactivated",
+            to_emails=[acting_user.delivery_email],
+            from_name=FromAddress.security_email_from_name(language=language),
+            from_address=FromAddress.tokenized_no_reply_address(),
+            language=language,
+            context=acting_owner_context,
+            realm=realm,
+        )
+
+        # If there are other active organization owners, then they
+        # receive an email that includes the deactivating owner's name.
+        other_owners_context = shared_context.copy()
+        other_owners_context["deactivating_owner"] = acting_user.full_name
+        other_owners_context["is_acting_owner"] = False
+        if owner_emails:
+            send_email(
+                "zerver/emails/realm_deactivated",
+                to_emails=list(owner_emails),
+                from_name=FromAddress.security_email_from_name(language=language),
+                from_address=FromAddress.tokenized_no_reply_address(),
+                language=language,
+                context=other_owners_context,
+                realm=realm,
+            )
```

Comment:
If we decide to turn this into a loop over owners, the logic might actually get simpler since we can do stuff like just set `is_acting_owner` in a context object constructed in each line.

You can probably also replace the `.copy` stuff with this pattern:
```
other_owners_context = dict(
    deactivating_owner=acting_user.full_name,
    ...
    **shared_context,
)
```

Reply:
Updates have us loop over the owners which is indeed simpler logic for setting the email context.